### PR TITLE
Fixed some Wicket CSP violations

### PR DESCRIPTION
**Pull request by sikeoka**
This PR should fix all of the remaining CSP violations from the Wicket 9 upgrade after https://github.com/geoserver/geoserver/pull/8053 is merged. I do not use any of the extensions or community modules so those changes are completely untested by me but I'm putting it here if anyone is interested.

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Review by mprins — APPROVED** [review]
lgtm (untested)

File: src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java [hunk 1]
```diff
@@ -449,11 +451,21 @@ public LayerPreviewPanel(String id, IModel<ImportTask> model) {
...
-            String javascript =
-                    "go(document.getElementById('" + get("links").getMarkupId() + "'));";
-            add(
-                    new ExternalLink("go", "#")
-                            .add(new AttributeModifier("onclick", new Model<>(javascript))));
+            add(new ExternalLink("go", "#").setOutputMarkupId(true));
+        }
+
+        @Override
+        public void renderHead(IHeaderResponse response) {
```
Comment (aaime) [inline src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java:458]: This one is not working. When importing multiple layers, the "go" links all make the first layer preview appear. E.g., I've imported the tasmania shapefiles in another workspace, after a successful import, all "go" link preview the cities layer:

image

File: src/community/elasticsearch/src/main/java/org/geoserver/elasticsearch/ElasticConfigurationPage.java [hunk 2]
```diff
@@ -293,12 +295,25 @@ protected Component getComponentForProperty(
                         } else if (property == ElasticAttributeProvider.USE) {
                             CheckBox checkBox =
                                     new CheckBox("use", new PropertyModel<>(itemModel, "use"));
-                            final String onclick =
-                                    "document.getElementById(\""
+                            checkBox.setOutputMarkupId(true);
```
Comment (aaime) [inline src/community/elasticsearch/src/main/java/org/geoserver/elasticsearch/ElasticConfigurationPage.java:298]: Manually tested, it works.

**Review by aaime — CHANGES_REQUESTED** [review]

File: src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java (hunk 1 above)
Comment (sikeoka) [reply src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java:458]: I looked at the code and I'm wondering if this behavior you're describing is actually the current behavior without this PR?

Comment (sikeoka) [reply src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java:458]: There's also still more inline script on this page that I need to remove.

Comment (aaime) [reply src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java:458]: > I looked at the code and I'm wondering if this behavior you're describing is actually the current behavior without this PR?

That may well be true, I haven't used the importer GUI in a long while (and did not test it before this change).

**Comment by sikeoka** [comment]
It turns out that that original commit missed a lot of inline styles that used style tags in the HTML page. I updated this PR to fix them all too. The CAS and OAUTH2 panels will inherit the correct style from `org.geoserver.security.web.auth.PreAuthenticatedUserNameFilterPanel`.
An inline script issue in GeoServerLoginPage was fixed. GeoServerBasePage already adds the `placeholder()` call so it was being called twice on the login page.

File: src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java (hunk 1 above)
Comment (aaime) [reply src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskTable.java:458]: Tested again and found it working... duh, not sure what happened here.

**Pull request merged**